pkg/cloudprovider/provider/vsphere/provider.go

```diff
@@ -271,7 +275,11 @@ func (p *provider) Validate(spec v1alpha1.MachineSpec) error {
 	if err != nil {
 		return fmt.Errorf("failed to get vsphere client: '%v'", err)
 	}
-	defer client.Logout(context.TODO())
+	defer func() {
+		if lerr := client.Logout(context.TODO()); lerr != nil {
+			glog.Errorf("vsphere client failed to logout: %s", lerr)
```

Comment:
Instead of logging those situations, we should call `utilruntime.HandleErr`
It also logs the error, but additionally increases the error-counter-metric

Reply:
Fixed